lib/spoom/snapshot.rb

```diff
@@ -0,0 +1,96 @@
+# typed: strict
+# frozen_string_literal: true
+
+module Spoom
+  class Snapshot < Hash
```

Comment:
My first try was using a typed struct.

But since the snapshot data are mainly used for display it means no looping over the fields etc without meta-programming.

I opted to use a hash instead to make it easier for the client.

Reply:
There is always `T::Struct#serialize` method to turn it into a Hash. That might still be better, don't you think?